### PR TITLE
Add huggingface_hub client library

### PR DESCRIPTION
**Pull request by FarukOzderim**
Add huggingface_hub client library and model comparator space builder section.

### CODE REVIEW

File: guides/using_hugging_face_integrations.md [hunk 1]
```diff
@@ -140,6 +140,31 @@ iface3.launch()
 
 Although both models are generative, you can see that the way both models behave is very different. That's a powerful application of `Parallel`!
 
+## Creating spaces with python
```
Comment (osanseviero) [inline guides/using_hugging_face_integrations.md:143]:
```suggestion
## Creating Spaces with python
```

File: guides/using_hugging_face_integrations.md [hunk 2]
```diff
@@ -140,6 +140,31 @@ iface3.launch()
...
+
+Making use of the [huggingface_hub client library](https://github.com/huggingface/huggingface_hub/tree/main/src/huggingface_hub) library you can create new spaces or models. You can do this even in a Gradio space! You can find an example space [here](https://huggingface.co/spaces/farukozderim/Model-Comparator-Space-Builder). This space creates a new space comparing different models or spaces with the support of Gradio `load` and `Parallel`. Now you can try creating cool spaces with all kinds of functionality 😎.
```
Comment (osanseviero) [inline guides/using_hugging_face_integrations.md:145]:
```suggestion
Making use of the [huggingface_hub client library](https://github.com/huggingface/huggingface_hub/tree/main/src/huggingface_hub) library you can create new Spaces or model repositories. You can do this even in a Gradio Space! You can find an example space [here](https://huggingface.co/spaces/farukozderim/Model-Comparator-Space-Builder). This Space creates a new Space comparing different models or spaces with the support of Gradio `load` and `Parallel`. Now you can try creating cool spaces with all kinds of functionality 😎.
```

File: guides/using_hugging_face_integrations.md [hunk 3]
```diff
@@ -140,6 +140,31 @@ iface3.launch()
...
+create_repo(name=target_space_name, token=hf_token, repo_type="space", space_sdk="gradio")
+repo_name = get_full_repo_name(model_id=target_space_name, token=hf_token)
+file_url = upload_file(
+    path_or_fileobj="file.txt",
+    path_in_repo="app.py",
+    repo_id=repo_name,
+    repo_type="space",
+    token=hf_token,
+)
+```
```
Comment (osanseviero) [inline guides/using_hugging_face_integrations.md:162]: very nice snippet!

File: guides/using_hugging_face_integrations.md [hunk 4]
```diff
@@ -140,6 +140,31 @@ iface3.launch()
...
+Here `create_repo` creates a gradio repo with the target name under a specific account using that account's Write Token. `repo_name` gets the full repo name of the related repo. Finally `upload_file` uploads a file inside the repo with the name `app.py`.
```
Comment (osanseviero) [inline guides/using_hugging_face_integrations.md:163]:
```suggestion
Here, `create_repo` creates a gradio repo with the target name under a specific account using that account's Write Token. `repo_name` gets the full repo name of the related repo. Finally `upload_file` uploads a file inside the repo with the name `app.py`.
```

**Review by osanseviero — APPROVED** [review]
Very nice, thanks!

File: guides/using_hugging_face_integrations.md (hunk 2 above)
Comment (osanseviero) [inline guides/using_hugging_face_integrations.md:145]: Maybe let's link to https://huggingface.co/docs/huggingface_hub/index instead

**Comment by FarukOzderim** [comment]
Thanks for the detailed review @osanseviero!

**Comment by aliabd** [comment]
LGTM

**Pull request merged**